src/PerformanceMonitor/PerformanceMonitor.NetworkProtocol.cs

```diff
@@ -0,0 +1,514 @@
+        private void OnRpcTimeCommand(string url)
+        {
+            if (!Uri.TryCreate(url, UriKind.Absolute, out var uri))
+            {
+                Console.WriteLine("Input url is invalid");
+                return;
+            }
+
+            try
+            {
+                var responseTime = GetRpcResponseTime(url);
+
+                if (responseTime > 0)
+                {
+                    Console.WriteLine($"RPC response time: {responseTime:0.##} milliseconds");
+                }
+            }
+            catch (FileNotFoundException)
+            {
+                // for this command it is required that the RpcClient plugin is installed
+                Console.WriteLine("Install RpcClient module to use the rpc time command.");
+            }
+        }
+
+        /// <summary>
+        /// Gets the time in milliseconds to receive the response of a rpc request
+        /// </summary>
+        /// <returns>
+        /// Returns the time of the response from the rpc request in milliseconds
+        /// </returns>
+        [RpcMethod]
+        public JObject GetRpcTime(JArray _params)
+        {
+            if (_params.Count != 1)
+            {
+                throw new RpcException(-32602, "Invalid params");
+            }
+            var url = _params[0].AsString();
+            if (!Uri.TryCreate(url, UriKind.Absolute, out var uri))
+            {
+                throw new RpcException(-32602, "Invalid params");
+            }
+
+            try
+            {
+                var responseTime = GetRpcResponseTime(url);
+                if (responseTime <= 0)
+                {
+                    throw new RpcException(-32602, "Invalid params");
+                }
+
+                return responseTime;
+            }
+            catch (FileNotFoundException)
+            {
+                // for this command it is required that the RpcClient plugin is installed
+                throw new RpcException(-32500, "Application error");
+            }
+        }
+
+        /// <summary>
+        /// Gets the time to receive the response of a rpc request
+        /// </summary>
+        /// <param name="url">
+        /// The url of the rpc server
+        /// </param>
+        /// <returns>
+        /// Returns zero if any exception is thrown; otherwise, returns the time in milliseconds of
+        /// the response from the rpc request
+        /// </returns>
+        private long GetRpcResponseTime(string url, bool printMessages = false)
+        {
+            if (printMessages)
+            {
+                Console.WriteLine($"Sending a RPC request to '{url}'...");
+            }
+            bool hasThrownException = false;
+
+            RpcClient client = new RpcClient(url);
+            Stopwatch watch = Stopwatch.StartNew();
+
+            try
+            {
+                client.GetBlockCount();
```

Comment:
MaxTimeout? what's happend if the server never answer and it doesn't close the connection?

Reply:
It's handled by [`SendAsync`](https://github.com/simplitech/neo-plugins/blob/metrics-plugin/src/RpcClient/RpcClient.cs#L65-L67) method of RpcClient. `PostAsync` throws `HttpRequestException` on timeout. [ref](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.postasync?view=netcore-3.1#System_Net_Http_HttpClient_PostAsync_System_Uri_System_Net_Http_HttpContent_)

Reply:
By default it's 100 seconds https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.timeout?view=netcore-3.1 I think that it should be 30 or less

Reply:
I see your point.
I'm going to set the timeout in the method then.